### PR TITLE
[T4A7][W11-B3]Wong Kang Fei

### PR DESCRIPTION
**Pull request by kfwong**

### CODE REVIEW

**Comment by kfwong** [comment]
ready for review

File: test/java/seedu/addressbook/data/person/NameTest.java [hunk 1]
```diff
@@ -13,4 +13,10 @@
     public void setup() throws IllegalValueException{
         name = new Name("John Smith");
     }
+    
+    @Test(expected=NullPointerException.class)
```
Comment (okkhoy) [inline test/java/seedu/addressbook/data/person/NameTest.java]: cool! :+1: correct way to start!

File: src/seedu/addressbook/data/person/Name.java [hunk 2]
```diff
@@ -37,6 +37,10 @@ public static boolean isValidName(String test) {
         return test.matches(NAME_VALIDATION_REGEX);
     }
 
+    public boolean isSimilar(Name other){
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Name.java]: you can add the "unusual" cases like `null` test first before jumping in to a minimal valid case.

File: src/seedu/addressbook/data/person/Name.java [hunk 3]
```diff
@@ -38,7 +40,10 @@ public static boolean isValidName(String test) {
     }
 
     public boolean isSimilar(Name other){
-        return this.fullName.equals(other.fullName);
+        Pattern pattern = Pattern.compile("\\b(John|Smith|K)\\b");
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Name.java]: errr... why are you hard coding the name here? this is incorrect by all counts!

File: t [hunk 4]
```diff
@@ -0,0 +1,113 @@
+[1mdiff --git a/src/seedu/addressbook/data/person/Name.java b/src/seedu/addressbook/data/person/Name.java[m
```
Comment (okkhoy) [inline t]: what is this file????

**Comment by okkhoy** [comment]
@kfwong it seems like you have a good grasp of TDD :+1:  keep up the good work

Some comments added. Please ack & close the PR after reading comments.

File: t (hunk 4 above)
Comment (kfwong) [reply t:1]: Opps sorry I might have commited a redundant file! sorry about that!

**Pull request closed without merging**